src/Core/Configure/Engine/JsonConfig.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Core\Configure\Engine;
+
+use Cake\Core\Configure\ConfigEngineInterface;
+use Cake\Core\Configure\FileConfigTrait;
+use Cake\Core\Exception\Exception;
+use Cake\Core\Plugin;
+
+/**
+ * JSON engine allows Configure to load configuration values from
+ * files containing JSON strings.
+ */
+class JsonConfig implements ConfigEngineInterface
+{
+
+    use FileConfigTrait;
+
+    /**
+     * File extension.
+     *
+     * @var string
+     */
+    protected $_extension = '.json';
+
+    /**
+     * Constructor for JSON Config file reading.
+     *
+     * @param string|null $path The path to read config files from. Defaults to CONFIG.
+     */
+    public function __construct($path = null)
+    {
+        if ($path === null) {
+            $path = CONFIG;
+        }
+        $this->_path = $path;
+    }
+
+    /**
+     * Read a config file and return its contents.
+     *
+     * Files with `.` in the name will be treated as values in plugins. Instead of
+     * reading from the initialized path, plugin keys will be located using Plugin::path().
+     *
+     * @param string $key The identifier to read from. If the key has a . it will be treated
+     *   as a plugin prefix.
+     * @return array Parsed configuration values.
+     * @throws \Cake\Core\Exception\Exception When files don't exist or when
+     *   files contain '..' (as this could lead to abusive reads) or when there
+     *   is an error parsing the JSON string.
+     */
+    public function read($key)
+    {
+        $file = $this->_getFilePath($key, true);
+
+        $values = json_decode(file_get_contents($file), true);
+        if (json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
@lorenzo Done

@markstory `json_last_error_msg()` is only available since PHP 5.5. Should I add a function to basics.php like shown here http://php.net/manual/en/function.json-last-error-msg.php#113243 ?  It would be useful in showing specific error in JsonView too.